air/src/air/mod.rs

```diff
@@ -156,6 +156,9 @@ pub trait Air: Send + Sync {
     /// This could be any type as long as it can be serialized into a sequence of bytes.
     type PublicInputs: Serializable;
 
+    type Frame<E: FieldElement>: EvaluationFrame<E>;
+    type AuxFrame<E: FieldElement>: EvaluationFrame<E>;
```

Comment:
Are you trying to use generic associated types? because I get many errors of the type `error[E0658]: generic associated types are unstable` when trying to implement the `Air` traits.

Reply:
Yes, you'll need to compile using rust nightly. It sounds like we might wait to merge this in until it is stabilized (according to @irakliyk this should happen within the next couple months). I should note that while the code does compile right now, it's still incomplete and not properly handling row offsets and frame shifts yet. Hoping to finish it up within the next week, and will let you know when it's ready for review.